### ISSUE TITLE
I would like to implement a perfect matching algorithm

### ISSUE THREAD

**Issue by dvberkel** [petgraph/petgraph#296]
### Summary
I would like to be able to detect if a graph has a perfect matching. With this ability I can answer related questions about graphs.

### Motivation
For a graph `G`, a [matching](https://en.wikipedia.org/wiki/Matching_(graph_theory)) is

>  a set of pairwise non-adjacent edges, none of which are loops; that is, no two edges share a common vertex. Furthermore it contains

A _perfect matching_ matches all vertices of a graph.

### Details

This would add an algorithm to the `algo` module.

Wikipedia has an entry on the [blossom algorithm](https://en.wikipedia.org/wiki/Blossom_algorithm), a polynomial time algorithm to return a maximal matching.

I am more than willing to implement this myself.

**Comment by dvberkel** [comment]
It this something that people are interested in?

Should this be included in this repository? Or should it be a crate on its own?

**Comment by gzsombor** [comment]
@dvberkel Have you had time to do it?

**Comment by rapodaca** [comment]
@gzsombor, there's a complete implementation in [Gamma](https://crates.io/crates/gamma).

**Comment by RaoulLuque** [comment]
Hey, is this not resolved in #400? :) @starovoid

**Comment by starovoid** [comment]
> Hey, is this not resolved in [#400](https://github.com/petgraph/petgraph/pull/400)? :) [@starovoid](https://github.com/starovoid)

Yes, it's resolved. We have an [algorithm](https://docs.rs/petgraph/latest/petgraph/algo/matching/fn.maximum_matching.html) and a [`Matching`](https://docs.rs/petgraph/latest/petgraph/algo/matching/struct.Matching.html) type with the `is_perfect` method.
